pkg/resources/catalog.cattle.io/v1/clusterrepo/validator_test.go

```diff
@@ -0,0 +1,119 @@
+package clusterrepo
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	catalogv1 "github.com/rancher/rancher/pkg/apis/catalog.cattle.io/v1"
+	"github.com/rancher/webhook/pkg/admission"
+	"github.com/stretchr/testify/assert"
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestClusterRepoValidation(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
```

Comment:
I would add a test case for when neither GitRepo, nor URL is set.

Reply:
Done